parser/src/main/kotlin/tools/samt/parser/Nodes.kt

```diff
@@ -5,6 +5,10 @@ import tools.samt.common.SourceFile
 
 sealed class Node(val location: Location)
 
+sealed interface AnnotatedNode {
+    val annotations: List<AnnotationNode>
+}
```

Comment:
A bit weird here because all the other base types are classes. I would've just implemented everything as `sealed interface` instead of `sealed class` to easily allow for this multi-inheritance, but this seems like the easiest implementation without a major refactoring